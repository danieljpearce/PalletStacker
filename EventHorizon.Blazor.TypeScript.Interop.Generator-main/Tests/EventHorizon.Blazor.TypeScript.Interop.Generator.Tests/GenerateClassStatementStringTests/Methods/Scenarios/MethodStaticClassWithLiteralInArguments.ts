declare module Static.Class {
    export class ExampleClass {
        static CreateGroundFromHeightMap(name: string, url: string, options: {
            width?: number;
            height?: number;
            subdivisions?: number;
            minHeight?: number;
            maxHeight?: number;
            colorFilter?: Color3;
            alphaFilter?: number;
            updatable?: boolean;
            onReady?: (mesh: GroundMesh) => void;
        }, scene?: Nullable<Scene>): GroundMesh;
        static ObjectLiteralAsSecondArgument(options: string | {
            value1?: number;
        }): void;
        static ObjectLiteral(options: object): void;
        static ObjectLiteralFirst(meshesOrMinMaxVector: {
            min: Vector3;
            max: Vector3;
        } | AbstractMesh[]): Vector3;
    }
}
