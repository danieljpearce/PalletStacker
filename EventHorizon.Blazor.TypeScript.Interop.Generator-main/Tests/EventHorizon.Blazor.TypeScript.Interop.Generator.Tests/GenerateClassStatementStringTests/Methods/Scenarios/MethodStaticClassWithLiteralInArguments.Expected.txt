/// Generated - Do Not Edit
namespace Static.Class
{
    using System;
    using System.Collections.Generic;
    using System.Text.Json.Serialization;
    using System.Threading.Tasks;
    using EventHorizon.Blazor.Interop;
    using EventHorizon.Blazor.Interop.Callbacks;
    using EventHorizon.Blazor.Interop.ResultCallbacks;
    using Microsoft.JSInterop;

    
    
    [JsonConverter(typeof(CachedEntityConverter<ExampleClass>))]
    public class ExampleClass : CachedEntityObject
    {
        #region Static Accessors

        #endregion

        #region Static Properties

        #endregion

        #region Static Methods
        public static GroundMesh CreateGroundFromHeightMap(string name, string url, object options, Scene scene = null)
        {
            return EventHorizonBlazorInterop.FuncClass<GroundMesh>(
                entity => new GroundMesh() { ___guid = entity.___guid },
                new object[]
                {
                    new string[] { "Static", "Class", "ExampleClass", "CreateGroundFromHeightMap" }, name, url, options, scene
                }
            );
        }

        public static void ObjectLiteralAsSecondArgument(string options)
        {
            EventHorizonBlazorInterop.Func<CachedEntity>(
                new object[]
                {
                    new string[] { "Static", "Class", "ExampleClass", "ObjectLiteralAsSecondArgument" }, options
                }
            );
        }

        public static void ObjectLiteral(object options)
        {
            EventHorizonBlazorInterop.Func<CachedEntity>(
                new object[]
                {
                    new string[] { "Static", "Class", "ExampleClass", "ObjectLiteral" }, options
                }
            );
        }

        public static Vector3 ObjectLiteralFirst(AbstractMesh[] meshesOrMinMaxVector)
        {
            return EventHorizonBlazorInterop.FuncClass<Vector3>(
                entity => new Vector3() { ___guid = entity.___guid },
                new object[]
                {
                    new string[] { "Static", "Class", "ExampleClass", "ObjectLiteralFirst" }, meshesOrMinMaxVector
                }
            );
        }
        #endregion

        #region Accessors

        #endregion

        #region Properties

        #endregion
        
        #region Constructor
        public ExampleClass() : base() { }

        public ExampleClass(
            ICachedEntity entity
        ) : base(entity)
        {
            ___guid = entity.___guid;
        }


        #endregion

        #region Methods

        #endregion
    }
}